processing/src/main/java/io/druid/query/PrioritizedExecutorService.java

```diff
@@ -50,9 +51,10 @@ public static PrioritizedExecutorService create(Lifecycle lifecycle, ExecutorSer
             config.getNumThreads(),
             0L,
             TimeUnit.MILLISECONDS,
-            new PriorityBlockingQueue<Runnable>(),
+            new PriorityBlockingQueue<Runnable>(config.getNumThreads()),
```

Comment:
this probably isn't doing that much- it's likely to be exceeded as soon as the user has more segments than processing threads